go-controller/pkg/util/net_linux.go

```diff
@@ -274,74 +278,94 @@ func LinkRoutesDel(link netlink.Link, subnets []*net.IPNet) error {
 }
 
 // LinkRoutesAdd adds a new route for given subnets through the gwIPstr
-func LinkRoutesAdd(link netlink.Link, gwIP net.IP, subnets []*net.IPNet, mtu int) error {
+func LinkRoutesAdd(link netlink.Link, gwIP net.IP, subnets []*net.IPNet, mtu int, src net.IP) error {
 	for _, subnet := range subnets {
 		route := &netlink.Route{
 			Dst:       subnet,
 			LinkIndex: link.Attrs().Index,
 			Scope:     netlink.SCOPE_UNIVERSE,
 			Gw:        gwIP,
 		}
+		if len(src) > 0 {
+			route.Src = src
+		}
```

Comment:
I am ok with this, but it just occurred to me that if you pass `nil` as `src` we don't really need this explicit check as Route.src is initialized to nil anyway. So using this would have the same effect:
```
route := &netlink.Route{
			Dst:       subnet,
			LinkIndex: link.Attrs().Index,
			Scope:     netlink.SCOPE_UNIVERSE,
			Gw:        gwIP,
                        Src:        src,
		}
```
Of course, assuming that an empty (but non-nil) byte array would not be passed as src to this method, but I don't think that's really possible if the net.IP methods are being used to build src.

Anyway, I am ok with how this is now so I will leave it up to you.

Reply:
OK, let's leave it as is then.